src/types/types.ts

```diff
@@ -0,0 +1,30 @@
+enum InputType {
+  'text',
+  'email',
+  'password',
+  'tel',
+  'number',
+}
```

Comment:
Попробуй тут использовать discriminated union
```suggestion
type InputType = {
  'text',
  'email',
  'password',
  'tel',
  'number',
}
```